tracer/src/Datadog.Trace.Tools.Runner/Checks/GacCheck.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="GacCheck.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Spectre.Console;
+using static Datadog.Trace.Tools.Runner.Checks.Resources;
+
+namespace Datadog.Trace.Tools.Runner.Checks
+{
+    internal class GacCheck
+    {
+        public static bool Run()
+        {
+            if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            {
+                return true;
+            }
+
+            var gacFolder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Microsoft.NET", "assembly", "GAC_MSIL", "Datadog.Trace");
+
+            if (!Directory.Exists(gacFolder))
+            {
+                Utils.WriteError(MissingGac);
+                return false;
+            }
+
+            bool foundVersions = false;
+
+            foreach (var folder in Directory.GetDirectories(gacFolder))
+            {
+                if (!File.Exists(Path.Combine(folder, "Datadog.Trace.dll")))
+                {
+                    continue;
+                }
+
+                var name = new DirectoryInfo(folder).Name;
+
+                // Format: v4.0_2.3.0.0__def86d061d0d2eeb
+                var match = Regex.Match(name, @"v4.0_(?<version>\d+\.\d+\.\d+\.\d+)__def86d061d0d2eeb");
+
+                var version = match.Success ? match.Groups["version"].Value : name;
+
+                AnsiConsole.WriteLine(GacVersionFormat(version));
+                foundVersions = true;
```

Comment:
I wander, can we flag a mismatched version here? 🤔 e.g. can we find a Datadog.Trace in the bin folder with a version that doesn't match the GACed version?

Reply:
I don't think there's much value doing this kind of check here. If you have multiple versions of Datadog.Trace in the GAC you can't predict which one is going to be loaded.

**But** now that you mention it, in the process check I should detect cases where multiple versions of Datadog.Trace.dll are loaded and log a warning. I will add that in a separate PR.